scale-decode/src/error/linkedlist.rs

```diff
@@ -0,0 +1,68 @@
+// Copyright (C) 2023 Parity Technologies (UK) Ltd. (admin@parity.io)
+// This file is a part of the scale-value crate.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//         http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::sync::Arc;
+
+// A simple singly-linked `Send`-able linked list
+// implementation to allow fairly cheap path cloning
+// and appending. This is used as part of our `Context`
+// type and is not a part of the public API.
+#[derive(Debug, Clone)]
+pub struct LinkedList<T>(Option<Arc<Inner<T>>>);
```

Comment:
I don't really understand why you need a `LinkedList` here, isn't `VecDeque` sufficient?

It seems like it's just need `push_front` and iterator over it?

Reply:
Maybe you're right! I wrote the linkedlist back when I was passing context in, and so needed to be abel to cheaply clone and add items to the front. But now, since it's just appended to on the way back, you're totally right and a simple Vec should do the trick!

Reply:
Got rid of LinkedList and replaced with a Vec; I'll do the same in scale-encode too!

Reply:
https://github.com/paritytech/scale-encode/pull/5